src/pybind/mgr/alert/module.py

```diff
@@ -0,0 +1,194 @@
+        self.config['twitter_consumer_secret'] = \
+            int(self.get_config("twitter_consumer_key", default=self.config_keys['twitter_consumer_key']))
+        self.config['twitter_access_token'] = \
+            int(self.get_config("twitter_access_token", default=self.config_keys['twitter_access_token']))
+        self.config['twitter_access_token_secret'] = \
+            int(self.get_config("twitter_access_token_secret", default=self.config_keys['twitter_access_token_secret']))
+
+    def get_api():
+        auth = tweepy.OAuthHandler(self.config['twitter_consumer_key'], self.config['twitter_consumer_secret'])
+        auth.set_access_token(slef.config['twitter_access_token'], self.config['twitter_access_token_secret'])
+        return tweepy.API(auth)
+
+    def tweets(alert_message):
+        api = get_api()
+        tweet = alert_message
+        status = api.update_status(status=tweet)
+        return status
+
+    def slack(alert_message):
+        slack = Slacker(self.config['slack_api'])
+        post = alert_message
+        return post
+
+    def email(alert_message):
+        msg = MIMEText(alert_message)
+        msg['Subject'] = 'Ceph: Cluster Health Status Alert'
+        msg['From'] = self.config['email_sender']
+        msg['To'] = self.config['email_receiver']
+        server = smtplib.SMTP('{}:{}'.format(self.config['mail_server'], self.config['mail_port']))
+        server.ehlo()
+        server.starttls()
+        server.login(self.config['email_sender'], self.config['sender_password'])
+        sendmail = server.sendmail(self.config['email_sender'], self.config['email_receiver'],msg.as_string())
+        return sendmail
+
+    def new_alert(self, cmd):
+        self.refresh_config()
+        if option in ['slack_api']:    
+
+    def handle_command(self, inbuf, cmd):
+        self.log.error("handle_command")
+        if cmd['prefix'] == 'email alert':
+            return self.alert_via_email(cmd)
+        elif cmd['prefix'] == 'slack alert':
+            return self.post_to_slack(cmd)
+        else:
+            raise NotImplementedError(cmd['prefix'])
+
+    def refresh_config(self):
+        for opt in self.OPTIONS:
+            setattr(self,
+                    opt['name'],
+                    self.get_config(opt['name']) or opt['default'])
+            self.log.debug(' %s = %s', opt['name'], getattr(self, opt['name']))
+
+    def serve(self):
+        """
+        This method is called by the mgr when the module starts and can be
+        used for any background activity.
+        """
+        self.log.info("Starting")
+        old_health = None
+        old_health = json.loads(self.get('health')['json'])
+
+        while self.run:
+            self.refresh_config()
+            if time.time() >= time_last_checked:
+                time_last_checked = time.time()+60
+                new_health = json.loads(self.get('health')['json'])
+                for code, item in new_health['checks'].iteritems():
```

Comment:
before this line you should add
```
new = {}
old = {}
updated = {}
```
so that you're populating an existing dictionary.